features/lorawan/lorastack/phy/LoRaPHY.cpp

```diff
@@ -629,30 +634,36 @@ void LoRaPHY::restore_default_channels()
 
 bool LoRaPHY::verify_rx_datarate(uint8_t datarate)
 {
-    if (phy_params.dl_dwell_time_setting == 0) {
-        //TODO: Check this! datarate must be same as minimum! Can be compared directly if OK
-        return val_in_range(datarate,
-                            phy_params.min_rx_datarate,
-                            phy_params.min_rx_datarate);
-    } else {
-        return val_in_range(datarate,
-                            phy_params.dwell_limit_datarate,
-                            phy_params.min_rx_datarate );
+    if (is_datarate_supported(datarate)) {
+        if (phy_params.dl_dwell_time_setting == 0) {
+            //TODO: Check this! datarate must be same as minimum! Can be compared directly if OK
+            return val_in_range(datarate,
+                                phy_params.min_rx_datarate,
+                                phy_params.max_rx_datarate);
+        } else {
+            return val_in_range(datarate,
+                                phy_params.dwell_limit_datarate,
+                                phy_params.max_rx_datarate );
+        }
     }
+    return false;
 }
 
 bool LoRaPHY::verify_tx_datarate(uint8_t datarate, bool use_default)
 {
-    if (use_default) {
-        return val_in_range(datarate, phy_params.default_datarate,
-                            phy_params.default_max_datarate);
-    } else if (phy_params.ul_dwell_time_setting == 0) {
-        return val_in_range(datarate, phy_params.min_tx_datarate,
-                            phy_params.max_tx_datarate);
-    } else {
-        return val_in_range(datarate, phy_params.dwell_limit_datarate,
-                            phy_params.max_tx_datarate);
+    if (is_datarate_supported(datarate)) {
```

Comment:
Don't need to change it this time, but general style suggestion - use early exit for parameter validation

    if (!is_datarate_supported(datarate)) {
        return false;
    }

Saves indenting all the body of the function, reducing diffs and merge conflicts. This style doesn't scale well as the number of parameter checks increases.